### ISSUE TITLE
Syncing issue — two different instances of the same page

### ISSUE THREAD

**Issue by jdwds** [Roam-Research/issues#265]
I noticed that a page I was working on last night using my laptop hadn't synced with Roam on my desktop, the desktop version was appearing as an older instance of the page. The pages both apparently were synced (the indicator was green and the last change was recent). I tried refreshing/restarting etc. but nothing changed on the laptop/desktop . I checked Roam on my phone and the desktop version was the one stored on the server so it seems there was a clash between the one cached on my laptop and server. I fixed it by deleting the website data from Chrome on my laptop and it pulled down the correct page (I backed up the cached version since that's the one I actually wanted).
I can't reproduce it now so maybe it was just a one off but thought I'd let you guys know!

** System information **
 - Roam version: 0.6.2-db48078f
 - Device: laptop is an MacBook and desktop is an iMac
 - OS: macOS Catalina
 - Browser: both Chrome

**Comment by mg0716** [comment]
This one might be related to https://github.com/Roam-Research/issues/issues/201

**Comment by jdwds** [comment]
It's happened again but I've not tried to fix it. Here are some screenshots, the problem is the in the definition of a simple function block:

This is on my MacBook (wrong):
<img width="818" alt="image" src="https://user-images.githubusercontent.com/43283366/82906905-d9054e80-9f5d-11ea-989c-945acb80b06e.png">

This is on my iMac (correct):
![image](https://user-images.githubusercontent.com/43283366/82907054-07832980-9f5e-11ea-85b8-f05d5b3d1828.png)

The correct one is the iMac (the one that's also stored on the server). It's strange because I inputted everything using the iMac (with the MacBook asleep) by first pasting in the text then I edited it to make sure the LaTeX rendered correctly; it seems the server synced once when I pasted it in, then again when I finished editing it. Then somehow when it opened my MacBook  and Roam synced it spliced the two instances together.

**Comment by filipesilva** [comment]
https://github.com/Roam-Research/issues/issues/49#issuecomment-634987849 lists another instance of this problem.

Still not sure what actually causes it but it seems that the cached state somehow misses things.

A few questions on the topic please:

were you working on multiple tabs simultaneously at the time?
were you low on disk space?

**Comment by jdwds** [comment]
@filipesilva 
> * were you working on multiple tabs simultaneously at the time?

So I normally have Roam open constantly on my MacBook and iMac but I've almost never got them both awake at the same time (I'm not sure if Roam is able to sync in the background whilst asleep or not, probably not on my MacBook but potentially on my iMac). In this instance I worked exclusively on the iMac until I'd finished taking notes then put it to sleep, then opened my MacBook and there found the problem.

> * were you low on disk space?

No I'm not on either machine. Also this has happened across my devices, i.e I've had this problem whether working on my MacBook and syncing to the iMac, or working on the iMac and syncing to the MacBook so I guess that would rule out certain things. 

---
As an update I'm guessing the MacBook pushed the problematic blocks to the server because the error has propagated through my devices now, so everything is in sync (I've corrected the error though).

**Comment by filipesilva** [comment]
@jdwds did this happen in your daily logs? If so, have you seen it happen outside the daily logs?

**Comment by jdwds** [comment]
@filipesilva no, none of this has happened on my daily logs. The daily logs haven't been a problem that I've known of.
